cloud-gateway-service/src/test/java/org/zowe/apiml/cloudgatewayservice/x509/X509UtilTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.cloudgatewayservice.x509;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.server.reactive.SslInfo;
+
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class X509UtilTest {
+
+    @Mock
+    private SslInfo sslInfo;
+
+    @Mock
+    private X509Certificate certificate;
+
+    private final X509Certificate[] x509Certificates = new X509Certificate[1];
+
+    @Test
+    void givenCorrectCertificate_whenGetCertificate_thenGetCertificate() throws CertificateEncodingException {
```

Comment:
Could you maybe use Nested annotation for better readibility? Also for the other test classes, to make it more granular.

Reply:
I don't think this class requires @Nested classes, but honestly, I am not against it.